### PR TITLE
✨ Gui - World Implementation - 1515028240

### PR DESCRIPTION
**Pull request by BxptisteM**
# Implementation of the GUI World class :

- Added a Position class (usefull to manipulate the positions of the objects)
- Fixed the object class to make it use the position class
- Added the world class with his methods
- Added tests

### CODE REVIEW

File: gui/src/position/Position.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+using namespace GUI;
+
+Position::Position(unsigned int x, unsigned int y)
+    :   _x(x)
+    ,   _y(y)
+{
+}
+
+Position::~Position()
```
Comment (RenardFute) [inline gui/src/position/Position.cpp:18]: could have been ``= default``

File: gui/src/world/World.hpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
+/*
+** EPITECH PROJECT, 2024
+** Zappy
+** File description:
+** World
+*/
+
+#ifndef WORLD_HPP_
+#define WORLD_HPP_
```
Comment (RenardFute) [inline gui/src/world/World.hpp:9]: Missing indentation 👀

**Review by RenardFute — CHANGES_REQUESTED** [review]
I feel like your include could use non-relative paths, since everything should be synced by the CMake. Otherwise, this seems nice to me, I just a constructor that accepts a position for every object and defaults it to 0, 0 would have been better since with the current system if you want to create an object at x, y it would take you 3 or more lines.

Fix the indentation error and ask me back ^^

**Review by RenardFute — APPROVED** [review]

**Review by 6im0n — APPROVED** [review]
good for me :+1:

File: gui/src/objects/eggs/Egg.cpp [hunk 3]
```diff
@@ -9,8 +9,8 @@
 
 using namespace GUI;
 
-Egg::Egg(std::string teamName, int ownerID)
-    :   AObject(0)
+Egg::Egg(std::string teamName, int ownerID, Position tile)
```
[the remark this answers is not in the document]
Comment (BxptisteM) [reply gui/src/objects/eggs/Egg.cpp:12]: This is the ID of the trantorian who laid the egg (I think we should display it when we click on the egg)

File: tests/gui/get_objects_positions_spec.cpp
[the remark this answers is not in the document]
Comment (BxptisteM) [reply tests/gui/get_objects_positions_spec.cpp:1]: I named my test files following the casing of the templates ones that where here at the very begining that's why but it's now fixed 🙂

File: gui/src/objects/eggs/Egg.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (Quentintnrl) [reply gui/src/objects/eggs/Egg.cpp:12]: We don't stock this data on the server

**Pull request merged**